src/PHPHtmlParser/Dom/HtmlNode.php

```diff
@@ -186,6 +186,9 @@ protected function clear()
         $this->innerHtml = null;
         $this->outerHtml = null;
         $this->text      = null;
+        if ( ! is_null($this->parent)) {
+            $this->parent->clear();
```

Comment:
Actually, looking at this change again I'm not sure this would be the ideal approach.

I think something more along to what I did in #101 would be better. I'll check if the current version of #101 fixes the issue here as is.